docs/structured-streaming-programming-guide.md

```diff
@@ -2452,6 +2452,14 @@ Specifically for built-in HDFS state store provider, users can check the state s
 it is best if cache missing count is minimized that means Spark won't waste too much time on loading checkpointed state.
 User can increase Spark locality waiting configurations to avoid loading state store providers in different executors across batches.
 
+#### State Data Source (Experimental)
+
+Spark provides the data source named "State Data Source" which provides the ability of manipulating states in the checkpoint. Users can run the batch query with State Data Source to get the visibility of the states for existing streaming query.
```

Comment:
Apache Spark provides a streaming state related data source that provides the ability to

Reply:
We don't really allow manipulating anything yet right ?

Reply:
IMO read ability could be considered as manipulation, and we are looking forward to address write ability as well, hence using manipulation does not sound incorrect to me.